### PR TITLE
Removed all obvious panic sources from verifier

### PR DESCRIPTION
**Pull request by pvienhage**

### CODE REVIEW

File: crypto/openstark/src/verifier.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@ pub fn check_proof<Public>(
     params: &ProofParams,
     trace_cols: usize,
     trace_len: usize,
-) -> bool
+) -> Result<(), &'static str>
```
Comment (z2trillion) [inline crypto/openstark/src/verifier.rs]: I think this should just return `Result<(), MerkleTreeError>`? You can then put the strings directly in the error, or use an enum, and derive Debug?

File: crypto/merkle-tree/src/result.rs [hunk 2]
```diff
@@ -10,3 +10,18 @@ pub enum Error {
     NotEnoughHashes,
     RootHashMismatch,
 }
+
+// Allows error bubbling up to string with '?' operator
```
Comment (Recmo) [inline crypto/merkle-tree/src/result.rs]: Is this standard? It seems wrong to convert errors into strings.

Comment (Recmo) [reply crypto/merkle-tree/src/result.rs]: Yes, this is an anti-pattern which we should not do. Instead we should implement the `Error` and `Display` traits.

https://doc.rust-lang.org/rust-by-example/error/multiple_error_types/define_error_type.html

https://doc.rust-lang.org/std/error/trait.Error.html

File: crypto/openstark/src/verifier.rs (hunk 1 above)
Comment (Recmo) [reply crypto/openstark/src/verifier.rs]: Verifier can have other errors than the MerkleTree ones, so we need nested errors here. Rust has standard ways of handling this that we should use.

**Pull request merged**